src/index.ts

```diff
@@ -169,13 +169,26 @@ export class Storage {
       for (const key of relevantKeyList) {
         callbackMap[key](
           {
-            newValue: JSON.parse(changes[key].newValue),
-            oldValue: JSON.parse(changes[key].oldValue)
+            newValue: this.#parseValue(changes[key].newValue),
```

Comment:
Can you update other JSON.parse values as well @louisgv

Reply:
wait where?

Reply:
Like in the other functions in the file that parse the storage val

Reply:
let's apply this for the whole file after we merge this imo.